app/ante.go

```diff
@@ -0,0 +1,156 @@
+		}
+
+		return anteHandlerFn(ctx, tx, simulate)
+	}
+}
+
+// newSigVerificationGasConsumer returns a SignatureVerificationGasConsumer that
+// returns zero gas fees for transactions which meet the following criteria:
+// - Has ONLY one signer
+// - Contains at least one message
+// - Contains ONLY morse claim messages (i.e. MsgClaimMorseAccount, MsgClaimMorseApplication, and MsgClaimMorseSupplier)
+func newSigVerificationGasConsumer(
+	sdkCtx cosmostypes.Context,
+	app *App,
+	tx cosmostypes.Tx,
+) ante.SignatureVerificationGasConsumer {
+	// Use the freeSecp256k1SigGasConsumer if the tx:
+	// - Has ONLY one signer
+	// - Contains at least one message
+	// - Contains ONLY Morse claim message(s)
+	if txHasOneSecp256k1Signature(tx) &&
+		txHasOnlyMorseClaimMsgs(tx) {
+		return freeSigGasConsumer
+	}
+
+	// Use the default signature verification gas consumer if ANY
+	// non-morse-claim message type is included in the tx.
+	return ante.DefaultSigVerificationGasConsumer
+}
+
+// newTxFeeChecker returns a TxFeeChecker that returns zero gas fees for
+// transactions that contain ONLY morse claim messages (i.e. MsgClaimMorseAccount,
+// MsgClaimMorseApplication, and MsgClaimMorseSupplier).
+func newTxFeeChecker(
+	sdkCtx cosmostypes.Context,
+	app *App,
+	tx cosmostypes.Tx,
+) ante.TxFeeChecker {
+	if txHasOneSecp256k1Signature(tx) &&
+		txHasOnlyMorseClaimMsgs(tx) {
+		return freeTxFeeChecker
+	}
+
+	// When nil is returned, ante.checkTxFeeWithValidatorMinGasPrices is used by
+	// default by cosmos-sdk.
+	// See:
+	// - https://github.com/cosmos/cosmos-sdk/blob/v0.50.13/x/auth/ante/fee.go#L31
+	// - https://github.com/cosmos/cosmos-sdk/blob/v0.50.13/x/auth/ante/ante.go#L48
+	return nil
+}
+
+// txHasOneSecp256k1Signature returns true if the given transaction contains
+// only one secp256k1 signature. If an error occurs while parsing the tx, or
+// retrieving the signer public key, it returns false (to fail safely).
+func txHasOneSecp256k1Signature(tx cosmostypes.Tx) bool {
+	sigTx, ok := tx.(authsigning.SigVerifiableTx)
+	if !ok {
+		return false
+	}
+
+	pubKeys, err := sigTx.GetPubKeys()
+	if err != nil {
+		return false
+	}
+
+	// Ensure that the transaction has only one signer.
+	if len(pubKeys) != 1 {
+		return false
+	}
+
+	// Check if the signer's public key  is a secp256k1 public key.
+	switch pubKeys[0].(type) {
+	case *secp256k1.PubKey:
+		return true
+	default:
+	}
+
+	return false
+}
+
+// txHasOnlyMorseClaimMsgs returns true if the given transaction contains ONLY
+// morse claim messages (i.e. MsgClaimMorseAccount, MsgClaimMorseApplication, and
+// MsgClaimMorseSupplier).
+func txHasOnlyMorseClaimMsgs(tx cosmostypes.Tx) bool {
+	msgs := tx.GetMsgs()
+	if len(msgs) == 0 {
+		return false
+	}
+
+	for _, msg := range msgs {
+		msgTypeUrl := cosmostypes.MsgTypeURL(msg)
+		switch msgTypeUrl {
+		case claimMorseAcctMsgTypeUrl,
+			claimMorseAppMsgTypeUrl,
+			claimMorseSupplierMsgTypeUrl:
+			// check the remaining messages...
```

Comment:
TODO? Which ones?

Reply:
No todo, this describes the `continue` branch's behavior.